clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy

```diff
@@ -109,4 +87,31 @@ class CreateAzureServerGroupAtomicOperation implements AtomicOperation<Map> {
 
     [serverGroups: [(description.region): [name: description.name]]]
   }
+
+  String createSubnet(String resourceGroupName, String virtualNetworkName) {
```

Comment:
This particular logic probably warrants a unit test or two that cover the various edge cases you've added as TODOs.